src/api/spec/controllers/labels_controller_spec.rb

```diff
@@ -0,0 +1,90 @@
+RSpec.describe LabelsController do
+  let(:user) { create(:confirmed_user, :with_home) }
+  let(:admin_user) { create(:admin_user, :with_home) }
+  let(:label_template) { create(:label_template, project: admin_user.home_project) }
+  let(:package) { create(:package, project: admin_user.home_project) }
+  let!(:label_admin) { create(:label, label_template: label_template, labelable: package) }
+
+  before do
+    Flipper.enable(:labels, user)
+    Flipper.enable(:labels, admin_user)
+  end
+
+  describe 'GET #index' do
+    render_views
+
+    context 'when labels are present' do
+      before do
+        login user
+
+        get :index, params: { project: admin_user.home_project.name, package: package.name, format: 'xml' }
+      end
+
+      it 'returns a list of labels' do
+        expect(Xmlhash.parse(response.body)['count']).to eq('1')
+        expect(Xmlhash.parse(response.body)['label']['id'].to_i).to eq(label_admin.id)
```

Comment:
If this is a controller spec, it's enough to check that the `assigns(:labels)` returns the proper labels. If you wanna test the xml generation then you have to do it in a feature spec or similar